firebase-database/src/androidMain/kotlin/dev/gitlive/firebase/database/database.kt

```diff
@@ -188,8 +187,30 @@ actual class DatabaseReference internal constructor(
     actual suspend fun removeValue() = android.removeValue()
         .run { if(persistenceEnabled) await() else awaitWhileOnline() }
         .run { Unit }
-}
 
+    actual suspend fun <T> runTransaction(strategy: KSerializer<T>, transactionUpdate: (currentData: T) -> T): DataSnapshot {
+        val deferred = CompletableDeferred<DataSnapshot>()
+        android.runTransaction(object : Transaction.Handler {
+
+            override fun doTransaction(currentData: MutableData) =
+                Transaction.success(transactionUpdate(decode(strategy, currentData)) as MutableData)
```

Comment:
```suggestion
           override fun doTransaction(currentData: MutableData): Transaction.Result {
                currentData.value = currentData.value?.let {
                    transactionUpdate(decode(strategy, it))
                }
                return Transaction.success(currentData)
            }
```

I'm suggesting this change because I think transactions are not working. 
We need to work with `currentData.value`, not with `currentData` itself.
Also we need to tolerate `null` value as described in the [Firebase documentation](https://firebase.google.com/docs/database/android/read-and-write#save_data_as_transactions).
I'm happy to help to update and merge this PR but I'm not sure I'll be able to do the same in js and iOS.